docs/spec/other/token.md

```diff
@@ -0,0 +1,316 @@
+
+**Method** : `GET`
+
+**Auth required** : NO
+
+**Data constraints**
+
+```json
+{
+    "account": "[valid account encoded as hex]",
+}
+```
+
+**Data example**
+
+```json
+{
+    "account": "942BC57C8B9024C829627D559906D4668F4C873C",
+}
+```
+
+Success Response
+
+**Code** : `200 OK`
+
+**Content example**
+
+```json
+[{
+    "amount": 1000,
+    "denom":"atom"
+},
+{   
+    "amount":1337,
+    "denom":"photon"
+}]
+```
+
+Error Response
+
+**Condition** : Bad Account Data
+
+**Code** : `400 BAD REQUEST`
+
+**Content** :
+
+```json
+{
+    "errors": [
+        "Invalid Account"
+    ]
+}
+```
+
+### Sequence
+
+**URL** : `/sequence`
+
+**Method** : `GET`
+
+**Auth required** : NO
+
+**Data constraints**
+
+```json
+{
+    "account": "[valid account encoded as hex]",
+}
+```
+
+**Data example**
+
+```json
+{
+    "account": "942BC57C8B9024C829627D559906D4668F4C873C",
+}
+```
+
+Success Response
+
+**Code** : `200 OK`
+
+**Content example**
+
+```json
+{
+    sequence:1,
+}
+```
+
+Error Response
+
+**Condition** : Bad Account Data
+
+**Code** : `400 BAD REQUEST`
+
+**Content** :
+
+```json
+{
+    "errors": [
+        "Invalid Account"
+    ]
+}
+```
+
+
+### Transfer
+
+This method is for creating transactions the transfer tokens between accounts.
+
+
+**URL** : `/transfer`
+
+**Method** : `POST`
+
+**Auth required** : NO
+
+**Signing procedure**: TODO
+
+**Data constraints**
+
+```json
+{
+    "account": "valid account encoded as hex",
+    "destination_account":"valid account encoded as hex",
+    "sequence": "Current sequence number for account",
+    "amount": "Integer amount of tokens to send",
+    "denom":"Name of token to send",
+    "pubkey": "Hex encoded public key for account",
+    "signature":"Signature under public key of the canoncalized data"
+}
+```
+
+**Data example**
+
+```json
+{
+    "account": "942BC57C8B9024C829627D559906D4668F4C873C",
+    "destination_account":"942BC57C8B9024C829627D559906D4668F4C873C",
+    "sequence":1,
+    "amount":1000,
+    "denom":"atom",
+    "pubkey":"1624DE622008317FF1A0FC2EDFA2911DCBC8283D73B0F0581039D4D1FE314907EEDCF1ADF6",
+    "signature":
+}
+```
+
+Success Response
+
+**Code** : `200 OK`
+
+**Content example**
+
+```json
+{
+    "response":"Tx Ok",
+}
+```
+
+Error Response
+
+**Condition** : Invalid Signature
+
+**Code** : `400 BAD REQUEST`
+
+**Content** :
+
+```json
+{
+    "errors": [
+        "Invalid Account"
+    ]
+}
+```
+
+Error Response
+
+**Condition** : Invalid Signature
+
+**Code** : `400 BAD REQUEST`
+
+**Content** :
+
+```json
+{
+    "errors": [
+        "Invalid Singature"
+    ]
+}
+```
+
+**Condition** : Insuffcient Funds
+
+**Code** : `400 BAD REQUEST`
+
+**Content** :
+
+```json
+{
+    "errors": [
+        "Insufficient funds"
+    ]
+}
+```
+
+**Condition** : Invalid Public Key
+
+**Code** : `400 BAD REQUEST`
+
+**Content** :
+
+```json
+{
+    "errors": [
+        "Invalid public keys"
+    ]
+}
+```
+
+### TotalSupply
+
+**URL** : `/transfer`
```

Comment:
is this a dupe or why does is have the same endpoint as here https://github.com/cosmos/cosmos-sdk/pull/1028/files#diff-f344ca7e4930d2f8366727867786622aR130 ?